src/main/java/com/cburch/logisim/gui/log/Signal.java

```diff
@@ -92,8 +92,8 @@ public void extend(Value v, long duration) {
       final var dur2 = new long[c + 1][];
       System.arraycopy(val, 0, val2, 0, c);
       System.arraycopy(dur, 0, dur2, 0, c);
-      val2[c + 1] = new Value[maxSize == 0 || (maxSize - cap) > CHUNK ? CHUNK : (maxSize - cap)];
-      dur2[c + 1] = new long[maxSize == 0 || (maxSize - cap) > CHUNK ? CHUNK : (maxSize - cap)];
+      val2[c] = new Value[maxSize == 0 || (maxSize - cap) > CHUNK ? CHUNK : (maxSize - cap)];
+      dur2[c] = new long[maxSize == 0 || (maxSize - cap) > CHUNK ? CHUNK : (maxSize - cap)];
```

Comment:
This change is not obvious, please explain cause of the issue and how this change fixes it.